packages/core/src/NumericInput/NumericInput.tsx

```diff
@@ -0,0 +1,95 @@
+import { forwardRef, useMemo, useRef } from "react";
+import { useForkRef } from "@mui/material";
+import {
+  ExtractNames,
+  useDefaultProps,
+} from "@hitachivantara/uikit-react-utils";
+
+import { HvAdornment } from "../FormElement";
+import { HvIcon } from "../icons";
+import { HvInput, HvInputProps } from "../Input";
+import { staticClasses, useClasses } from "./NumericInput.styles";
+
+export { staticClasses as numericInputClasses };
+
+export type HvNumericInputClasses = ExtractNames<typeof useClasses>;
+
+export interface HvNumericInputProps
+  extends Omit<HvInputProps<HTMLInputElement>, "inputProps"> {
+  classes?: HvNumericInputClasses;
+  /** Properties passed on to the input element. */
+  inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+}
```

Comment:
We should move `inputProps` to `HvBaseInput`. The problem with MUI's is that isn't the `HTMLInputElement | HTMLTextArea` elements, but the fact that is extends `React.HTMLAttributes` and not `React.InputHTMLAttributes`. Since `React.InputHTMLAttributes` itself extends `React.HTMLAttributes`, it's a safe/compatible change.

Also, we should be able to pass other `HvInputClasses` classes, instead of being limited to `HvNumericInputClasses`

```suggestion
export interface HvNumericInputProps extends HvInputProps {}
```